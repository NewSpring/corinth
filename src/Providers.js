import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { AuthProvider } from '@apollosproject/ui-auth';
import { AnalyticsProvider } from '@apollosproject/ui-analytics';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { NotificationsProvider } from '@apollosproject/ui-notifications';
import { LiveProvider } from '@apollosproject/ui-connected';

import NavigationService from './NavigationService';
import ClientProvider from './client';
import ExternalLinkProvider from './linking/Provider';
import track from './analytics';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider {...props}>
    <NotificationsProvider
      oneSignalKey={ApollosConfig.ONE_SIGNAL_KEY}
      navigate={NavigationService.navigate}
    >
      <AuthProvider
        navigateToAuth={() => NavigationService.navigate('Auth')}
        navigate={NavigationService.navigate}
        closeAuth={() => NavigationService.navigate('Onboarding')}
      >
        <ExternalLinkProvider navigate={NavigationService.navigate}>
          <MediaPlayerProvider>
            <AnalyticsProvider trackFunctions={[track]}>
              <LiveProvider>
                <Providers
                  themeInput={customTheme}
                  iconInput={customIcons}
                  {...props}
                />
              </LiveProvider>
            </AnalyticsProvider>
          </MediaPlayerProvider>
        </ExternalLinkProvider>
      </AuthProvider>
    </NotificationsProvider>
  </ClientProvider>
);

export default AppProviders;
