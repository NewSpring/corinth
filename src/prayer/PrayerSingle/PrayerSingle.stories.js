import React from 'react';
import { storiesOf } from '@apollosproject/ui-storybook';
import { PaddedView } from '@apollosproject/ui-kit';

import PrayerSingle from '.';

const prayer = {
  person: {
    photo: {
      uri: 'https://fillmurray.com/400/600',
    },
  },
  firstName: 'Bill',
  text:
    'I want to go back to school for my PTA degree next year, but I’m afraid of not getting accepted. I have a great job right now but I’m not very happy. Pray my plans for school align with God’s plans for me.',
  campus: { name: 'Anderson' },
};

storiesOf('prayer/PrayerSingle', module).add('Example', () => (
  <PaddedView>
    <PrayerSingle prayer={prayer} />
  </PaddedView>
));
