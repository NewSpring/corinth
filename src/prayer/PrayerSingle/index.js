import React, { memo } from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import moment from 'moment';
import {
  BodyText,
  H5,
  styled,
  Touchable,
  ChannelLabel,
} from '@apollosproject/ui-kit';
import { AnalyticsConsumer } from '@apollosproject/ui-analytics';
import PrayerHeader from '../PrayerHeader';

const GreyH5 = styled(({ theme }) => ({
  color: theme.colors.text.tertiary,
  marginBottom: theme.sizing.baseUnit * 0.5,
}))(H5);

const PrayerText = styled(({ theme }) => ({
  marginVertical: theme.sizing.baseUnit,
}))(BodyText);

const AbsolutePositionedView = styled(() => ({
  position: 'absolute',
  top: 0,
  right: 0,
  zIndex: 2,
}))(View);

const PrayerSingle = memo(
  ({
    showHelp,
    showHeader,
    showDate,
    avatarSize,
    prayer,
    action,
    ...props
  }) => (
    <View>
      <AbsolutePositionedView>{action}</AbsolutePositionedView>
      {showDate ? (
        <GreyH5>
          {prayer.startTime ? moment(prayer.startTime).fromNow() : ''}
        </GreyH5>
      ) : null}
      {showHeader ? (
        <PrayerHeader
          avatarSize={avatarSize}
          avatarSource={prayer.isAnonymous ? null : prayer.requestor.photo}
          title={`Pray for ${
            prayer.isAnonymous
              ? 'Request'
              : prayer.requestor.nickName || prayer.requestor.firstName
          }`}
          source={prayer.campus.name !== 'Web' ? prayer.campus.name : null}
        />
      ) : null}
      <PrayerText>{prayer.text}</PrayerText>
      {showHelp ? (
        <AnalyticsConsumer>
          {({ track }) => (
            <Touchable
              onPress={() => {
                props.navigation.navigate('ContentSingle', {
                  // TODO: this should come from a content channel
                  itemId: 'MediaContentItem:20f5b6548d64b1ac62a1c4b0deb0bfcb',
                  itemTitle: 'Learning how to pray like Jesus',
                  isolated: true,
                });
                track({ eventName: "Clicked 'How to Pray'" });
              }}
            >
              <View>
                <ChannelLabel
                  icon="information"
                  label="Not sure how to pray? Read this."
                />
              </View>
            </Touchable>
          )}
        </AnalyticsConsumer>
      ) : null}
    </View>
  )
);

PrayerSingle.propTypes = {
  showHelp: PropTypes.bool,
  showHeader: PropTypes.bool,
  showDate: PropTypes.bool,
  avatarSize: PropTypes.string,
  prayer: PropTypes.shape({}), // TODO: fill this out
  action: PropTypes.element,
};

PrayerSingle.defaultProps = {
  showHelp: false,
  showHeader: false,
  showDate: false,
  avatarSize: 'small',
  prayer: {
    firstName: 'Request',
    campus: { name: 'Web' },
    isAnonymous: false,
    person: { photo: { uri: '' } },
  },
  action: null,
};

PrayerSingle.displayName = 'PrayerSingle';

PrayerSingle.navigationOptions = {
  header: null,
};

export default PrayerSingle;
