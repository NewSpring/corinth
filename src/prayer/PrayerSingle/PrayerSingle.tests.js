import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'newspringchurchapp/src/Providers';

import PrayerSingle from '.';

const prayer = {
  person: {
    photo: {
      uri: 'https://fillmurray.com/400/600',
    },
  },
  firstName: 'Bill',
  text:
    'I want to go back to school for my PTA degree next year, but I’m afraid of not getting accepted. I have a great job right now but I’m not very happy. Pray my plans for school align with God’s plans for me.',
  campus: { name: 'Anderson' },
};

describe('the PrayerSingle component', () => {
  it('renders a prayer card', () => {
    const tree = renderer.create(
      <Providers>
        <PrayerSingle prayer={prayer} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
