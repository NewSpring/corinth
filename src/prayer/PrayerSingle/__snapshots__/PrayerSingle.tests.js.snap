// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the PrayerSingle component renders a prayer card 1`] = `
<View>
  <View
    style={
      Object {
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 2,
      }
    }
  />
  <Text
    bold={false}
    italic={false}
    style={
      Object {
        "color": "#303030",
        "fontFamily": "Colfax-Regular",
        "fontSize": 16,
        "lineHeight": 26,
        "marginVertical": 16,
      }
    }
  >
    I want to go back to school for my PTA degree next year, but I’m afraid of not getting accepted. I have a great job right now but I’m not very happy. Pray my plans for school align with God’s plans for me.
  </Text>
</View>
`;
